### DISCUSSION TITLE
Do you use Side View by Mozilla Firefox with Dark Reader?

### DISCUSSION THREAD

**Discussion by bershanskiy** [darkreader/darkreader#10105] [Polls]
**Update: Thanks for feedback. Dark Reader 4.9.59 will stop breaking sidebars (will leave them unchanged), Dark Reader 4.9.60 will support sidebars properly.**

[Side View by Mozilla Firefox](https://addons.mozilla.org/firefox/addon/side-view/) is an extension which allows user to open a page that would follow across the tabs:

![image](https://user-images.githubusercontent.com/45960703/195984643-2970eb9a-42f0-4c3e-8f88-baa6d5ca7ae9.png)
Image credit: [AMO](https://addons.mozilla.org/en-US/firefox/addon/side-view/)

Currently, Side View and similar extensions are not compatible with many extensions published on AMO, including Dark Reader. Unfortunately, Side View was last published on in September 2018, while the development officially ended with [the end of Firefox Test Pilot program in January 2019](https://medium.com/firefox-test-pilot/adios-amigo-51bec2a00072). Instead of fixing Side View, we can make a change within Dark Reader to work around this incompatibility. Unfortunately, change will add more complexity within Dark Reader and will effectively be more [technical debt](https://en.wikipedia.org/wiki/Technical_debt) within Dark Reader. I would like to asses whether this technical debt is worth the extra Firefox user convenience. In other words, whether sufficiently many users install Side View and Dark Reader at the same time.

Edit: this applies to all similar extension since they all use [`sidebarAction` API](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/user_interface/Sidebars). I updated this text, but can not update poll without resetting poll options. For purposes of the poll, please equate Side View to any other similar extension.

**Comment by Gusted** [comment]
Never knew we could make polls in discussions 😄. TIL

**Comment by marlop352** [comment]
This also applies to any sidebar add-on that I tested not just Mozilla's Side View.
There are more but those are the ones I tested when I first noticed this bug:

- https://addons.mozilla.org/en-US/firefox/addon/side-view/
- https://addons.mozilla.org/en-US/firefox/addon/tweetdeck-for-sidebar/
- https://addons.mozilla.org/en-US/firefox/addon/twitter-as-a-sidebar/
- https://addons.mozilla.org/en-US/firefox/addon/open-link-in-sidebar/

Probably most of the sidebar add-ons on AMO are either based on Side View or work similarly enough to be affected by this, and I would guess that they are as unmaintained as Side View.
I think that the most universal place to fix this problem right now would be on Dark Reader.

**Reply by bershanskiy** [reply]
Thanks for this comment, I updated the post to include all extensions using the same API.

**Comment by bershanskiy** [comment]
I can add support for Side View frames, but probably Dark Reader's ON/OFF state within iframes within Side View will be inconsistent with iframe state within regular tab (each iframe will be treated as a top-level frame for purposes of ON/OFF state). For example, if you enable Dark Reader globally and disable it on `sub.com` via site list feature and open `top.com` which frames `sub.com` within Side View and a regular tab, then `top.com` will be ON in both, but `sub.com` will be ON in tab and OFF in Side View. This inconsistency could be considered a bug, but fixing it would require some messaging visible to main wold so could in theory cause weird bugs.

**Comment by lfr** [comment]
@bershanskiy was the `4.9.60` change to "support sidebars properly" ever implemented? I can't find a way to enable dark reader on Firefox's Side View.

**Comment by AntonioRedondo** [comment]
[Side View](https://addons.mozilla.org/firefox/addon/side-view) was updated to version 0.6.6956 on 16 October 2024. Dark Reader 4.9.103 doesn't work on it though.
